src/lib/sql/sql_pipeline.hpp

```diff
@@ -14,6 +14,16 @@
 
 namespace opossum {
 
+// Holds relevant information about the execution of an SQLPipeline.
+struct SQLPipelineExecutionInfo {
```

Comment:
sorry to be that guy, but calling something "execution info" that contains info about many things (compiling, parsing, ...) that are NOT execution info ADDITIONALLY to info about the actual execution is counterintuitive to me. 
Just SQLPipeline{Statement}Timings would do it for me, I guess

Reply:
Well technically speaking this is info on the execution of the pipeline, which includes all steps from parsing to query execution. But if the name is bad, i can just change that. But Timings is also not that good, because it contains info on cache hit/miss too

Reply:
Sorry, missed the last part. Maybe just SQLPipelineMetrics then?

Reply:
yea. that sounds about right

Reply:
How about `SQLPipeline::Metrics`?

Reply:
I don't like nested structs that are supposed to be used outside of the structs. Too many colons that I need to type all the time.
No strong feelings though

Reply:
also not a big fan of the nested struct. I would be happy with `SQLPipeline{Statement}Metrics`

Reply:
Let's go for it.